### PR TITLE
Getting rid of build warnings

### LINKED ISSUE #90: Fix build warnings
Hi all!

I'm using this plugin to my ZF2 web application and I have one question. When I execute `npm run build` it shows to me some messages advising about a deprecated function. The message is the following:

```
Container#eachAtRule is deprecated. Use Container#walkAtRules instead.
Container#eachRule is deprecated. Use Container#walkRules instead.
Container#eachDecl is deprecated. Use Container#walkDecls instead.
Node#style() is deprecated. Use Node#raw()
Node#_value was deprecated. Use Node#raws.value
Hash: 02954ab3b14e71c533a7
Version: webpack 1.12.2
Time: 4923ms
         Asset    Size  Chunks             Chunk Names
tota11y.min.js  257 kB       0  [emitted]  app
    + 52 hidden modules

WARNING in tota11y.min.js from UglifyJs
Condition always true [./~/jquery/dist/jquery.js:9170,0]
Side effects in initialization of unused variable sourceMap [./~/style-loader/addStyles.js:185,0]
Side effects in initialization of unused variable media [./~/style-loader/addStyles.js:203,0]
Condition always false [./~/style-loader/addStyles.js:23,0]
Dropping unreachable code [./~/style-loader/addStyles.js:24,0]
Condition always false [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./less/tota11y.less:9,0]
Dropping unreachable code [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./less/tota11y.less:11,0]
Side effects in initialization of unused variable update [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./less/tota11y.less:7,0]
Condition always false [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/a11y-text-wand/style.less:9,0]
Dropping unreachable code [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/a11y-text-wand/style.less:11,0]
Side effects in initialization of unused variable update [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/a11y-text-wand/style.less:7,0]
Condition always false [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/contrast/style.less:9,0]
Dropping unreachable code [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/contrast/style.less:11,0]
Side effects in initialization of unused variable update [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/contrast/style.less:7,0]
Condition always false [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/headings/style.less:9,0]
Dropping unreachable code [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/headings/style.less:11,0]
Side effects in initialization of unused variable update [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/headings/style.less:7,0]
Condition always false [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/shared/annotate/style.less:9,0]
Dropping unreachable code [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/shared/annotate/style.less:11,0]
Side effects in initialization of unused variable update [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/shared/annotate/style.less:7,0]
Condition always false [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/shared/info-panel/style.less:9,0]
Dropping unreachable code [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/shared/info-panel/style.less:11,0]
Side effects in initialization of unused variable update [./~/style-loader!./~/css-loader!./~/postcss-loader!./~/autoprefixer-loader?{browsers:['> 1%']}!./~/less-loader!./plugins/shared/info-panel/style.less:7,0]

> tota11y@0.0.10 dev /var/www/dtome/guido/vendor/khan/tota11y
> webpack -d --devtool hidden --output-file=tota11y.js

output.file will be deprecated: Use 'output.filename' instead
Container#eachAtRule is deprecated. Use Container#walkAtRules instead.
Container#eachRule is deprecated. Use Container#walkRules instead.
Container#eachDecl is deprecated. Use Container#walkDecls instead.
Node#style() is deprecated. Use Node#raw()
Node#before is deprecated. Use Node#raws.before
Node#_value was deprecated. Use Node#raws.value
Hash: b8fcf00dab043270ca34
Version: webpack 1.12.2

```

There's something wrong with the plugin or the problem is in other way?

Thanks!

### PR DESCRIPTION
**Pull request by mveritym**
Fixes #90 
- [x] Removing autoprefixer (now part of postcss)
- [x] Updating postcss and postcss-loader to latest versions
- [x] Telling uglify to ignore warnings from node_modules

Feedback welcome, this is my first open source PR ever 🎉

### CODE REVIEW

**Comment by jdan** [comment]
Hey @mveritym,

Thanks so much for this - very excited to see you've made tota11y the target of your first open source PR!!

I'll give this a look today - but at a glance everything looks really great, this is going to be really valuable.

File: package.json [hunk 1]
```diff
@@ -32,7 +31,7 @@
   "scripts": {
     "build": "npm run prod && npm run dev",
     "prod": "webpack --config webpack.config.babel.js -p",
-    "dev": "webpack --config webpack.config.babel.js -d --devtool hidden --output-file=tota11y.js",
+    "dev": "webpack --config webpack.config.babel.js -d --devtool hidden --output=tota11y.js",
```
Comment (jdan) [inline package.json]: Looks like `tota11y.js` is no longer being generated because of this line. Is that possible?

File: package.json [hunk 2]
```diff
@@ -9,7 +9,6 @@
   "author": "Jordan Scales <scalesjordan@gmail.com>",
   "devDependencies": {
     "accessibility-developer-tools": "2.9.0-rc.0",
-    "autoprefixer-loader": "^2.0.0",
```
Comment (jdan) [inline package.json]: Do we need to configure autoprefixing within postcss now? It seems some properties like `align-items` are not being prefixed. Can you confirm? 

image

For reference, here's how `.tota11y-plugin-switch` is defined currently on master:

image

File: webpack.config.babel.js [hunk 3]
```diff
@@ -58,6 +58,7 @@ module.exports = {
         new webpack.ProvidePlugin({
             [options.jsxPragma]: path.join(__dirname, "utils", "element"),
         }),
+        new webpack.optimize.UglifyJsPlugin({compress: {warnings: false}})
```
Comment (jdan) [inline webpack.config.babel.js]: Great!

**Comment by jdan** [comment]
Apologies for the delayed review. I went on vacation and forgot to drop an update here that I was a bit behind schedule.

**Comment by mveritym** [comment]
@jdan no worries about the delay, hope you enjoyed your holiday! Fixed the pretty major stuff I missed, thanks for the review 🙂

File: webpack.config.babel.js [hunk 4]
```diff
@@ -58,6 +59,7 @@ module.exports = {
...
     ],
-    postcss: [veryimportant],
+    postcss: [veryimportant, autoprefixer],
```
Comment (jdan) [inline webpack.config.babel.js]: Looks like we'll need our browser config from above here. That is, `{browsers:['> 1%']}` removed from above on line 62 and passed into the `autoprefixer` plugin like so:

```
[veryimportant, autoprefixer({ browsers: ["> 1%"] })]
```

I think that will do the trick!

File: webpack.config.babel.js (hunk 3 above)
Comment (jdan) [inline webpack.config.babel.js]: Oh hmmm, I'm noticing that this is minifying `tota11y.js` too, but we want to keep that uncompressed.

I noticed this by running:
```
rm -rf build/
npm run build
```

and then viewing the contents of `build/tota11y.js`. Is there an easy way we can make this plugin only apply for the production case? (See `prod` in package.json's "scripts" field).

**Comment by jdan** [comment]
Hey @mveritym,

Just checking in - I'm going to be making a couple tweaks on here to fix the little nits I pointed out. Feel free to check out the commits that appear at the bottom of this page and comment as you see fit!

I just [fixed the issue](https://github.com/Khan/tota11y/pull/110/commits/6f277b7be4cb18d768626a7cfd7afab6c011e174) with the uglifyJS plugin usage (thanks to you for teaching me how we could include that in the babel config!) so that it conditionally added uglifyjs if we're in production mode.

**Comment by jdan** [comment]
Hey @mveritym,

I went ahead and squashed all your changes and [landed them on master](https://github.com/Khan/tota11y/pull/110/commits/5bda1a2570e856d03c73af2e6b12b8f5781abc09) ✨ 

Just wanted to give a special thanks for tackling this - dependency upgrades are difficult yet super important tasks that really help to make projects like tota11y stand the test of time. I really appreciate all the hard work you put into the PR, and I am honored that you chose tota11y as the home of your first open source pull request!

Feel free to reach out anytime if you're looking for another problem to tackle, or if you have any questions/comments on how we can make this project better for contributors. 

Take care, and thanks again!
@jdan

**Pull request merged**